___Execucao_com 2 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 3.24482
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:03
	Energy:	0.0001 kWh
	CO2eq:	0.0668 g
	This is equivalent to:
	0.0006 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:03
	Energy:	0.0001 kWh
	CO2eq:	0.0668 g
	This is equivalent to:
	0.0006 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 2 Treads_concluida___
___Execucao_com 4 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 1.78057
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:02
	Energy:	0.0001 kWh
	CO2eq:	0.0409 g
	This is equivalent to:
	0.0004 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:02
	Energy:	0.0001 kWh
	CO2eq:	0.0409 g
	This is equivalent to:
	0.0004 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 4 Treads_concluida___
___Execucao_com 6 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 1.29070
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0001 kWh
	CO2eq:	0.0325 g
	This is equivalent to:
	0.0003 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0001 kWh
	CO2eq:	0.0325 g
	This is equivalent to:
	0.0003 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 6 Treads_concluida___
___Execucao_com 8 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 1.09982
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0001 kWh
	CO2eq:	0.0287 g
	This is equivalent to:
	0.0003 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0001 kWh
	CO2eq:	0.0287 g
	This is equivalent to:
	0.0003 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 8 Treads_concluida___
___Execucao_com 10 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.87476
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0218 g
	This is equivalent to:
	0.0002 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 10 Treads_concluida___
___Execucao_com 12 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.77972
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 12 Treads_concluida___
___Execucao_com 14 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.61621
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 14 Treads_concluida___
___Execucao_com 16 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.52964
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 16 Treads_concluida___
___Execucao_com 18 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.62945
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 18 Treads_concluida___
___Execucao_com 20 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.55289
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 20 Treads_concluida___
___Execucao_com 22 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.58216
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 22 Treads_concluida___
___Execucao_com 24 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.72364
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 24 Treads_concluida___
___Execucao_com 26 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.59766
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 26 Treads_concluida___
___Execucao_com 28 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.55983
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 28 Treads_concluida___
___Execucao_com 30 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.55845
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 30 Treads_concluida___
___Execucao_com 32 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.50732
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 32 Treads_concluida___
___Execucao_com 34 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.48822
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 34 Treads_concluida___
___Execucao_com 36 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.46314
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 36 Treads_concluida___
___Execucao_com 38 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.43366
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 38 Treads_concluida___
___Execucao_com 40 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.55122
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 40 Treads_concluida___
___Execucao_com 42 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.41155
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 42 Treads_concluida___
___Execucao_com 44 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.42941
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 44 Treads_concluida___
___Execucao_com 46 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.44240
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 46 Treads_concluida___
___Execucao_com 48 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.38351
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 48 Treads_concluida___
___Execucao_com 50 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.42053
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 50 Treads_concluida___
___Execucao_com 52 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.43112
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 52 Treads_concluida___
___Execucao_com 54 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.43347
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 54 Treads_concluida___
___Execucao_com 56 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.58185
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 56 Treads_concluida___
___Execucao_com 58 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.44530
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 58 Treads_concluida___
___Execucao_com 60 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.46409
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 60 Treads_concluida___
___Execucao_com 62 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.61949
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 62 Treads_concluida___
___Execucao_com 64 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.40612
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 64 Treads_concluida___
___Execucao_com 66 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.39725
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 66 Treads_concluida___
___Execucao_com 68 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.53407
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 68 Treads_concluida___
___Execucao_com 70 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.41430
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 70 Treads_concluida___
___Execucao_com 72 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.44457
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 72 Treads_concluida___
___Execucao_com 74 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.46044
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 74 Treads_concluida___
___Execucao_com 76 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.43370
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 76 Treads_concluida___
___Execucao_com 78 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.44620
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 78 Treads_concluida___
___Execucao_com 80 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.43967
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 80 Treads_concluida___
___Execucao_com 82 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.68359
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 82 Treads_concluida___
___Execucao_com 84 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.48707
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 84 Treads_concluida___
___Execucao_com 86 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.41111
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:00
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 86 Treads_concluida___
___Execucao_com 88 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

Initialization:
unew_norm = 0.0509645
u_norm = 0.705265
error = 0.705287
-------------------------------

Converged after 18500 iterations
unew_norm: 0.412466
diff: 9.97582e-06
tolerance: 1e-05
error: 0.532394
Execution Time: 0.70956
CarbonTracker: WARNING - Epoch duration is too short for a measurement to be collected.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:01
	Energy:	0.0000 kWh
	CO2eq:	0.0000 g
	This is equivalent to:
	0.0000 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 88 Treads_concluida___
Os resultados foram salvos em log_result_1.txt
Os resultados foram salvos em log_result_2.txt
Os resultados foram salvos em log_result_3.txt
Os resultados foram salvos em log_result_4.txt
Os resultados foram salvos em log_result_5.txt
Os resultados foram salvos em log_result_6.txt
Os resultados foram salvos em log_result_7.txt
Os resultados foram salvos em log_result_8.txt
Os resultados foram salvos em log_result_9.txt
Os resultados foram salvos em log_result_10.txt
